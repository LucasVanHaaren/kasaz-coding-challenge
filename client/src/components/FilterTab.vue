<template>
  <v-expand-transition>
    <v-card outlined>
      <v-form>
        <v-container>
          <v-row>
            <v-col cols="12" align="center">
              <v-label>PRECIO</v-label>
            </v-col>
            <v-col>
              <v-row>
                <v-select
                  outlined
                  :items="filters.price"
                  label="min"
                  v-model="selectedFilters.minPrice"
                />
                <v-select
                  outlined
                  :items="maxPriceFilter"
                  label="max"
                  v-model="selectedFilters.maxPrice"
                />
              </v-row>
            </v-col>
          </v-row>

          <v-row>
            <v-col cols="12" align="center">
              <v-label>TAMAÑO</v-label>
            </v-col>
            <v-col>
              <v-row>
                <v-select
                  outlined
                  :items="filters.sqm"
                  label="min"
                  v-model="selectedFilters.minSqm"
                />
                <v-select
                  outlined
                  :items="maxSqmFilter"
                  label="max"
                  v-model="selectedFilters.maxSqm"
                />
              </v-row>
            </v-col>
          </v-row>

          <v-row>
            <v-col cols="12" align="center">
              <v-label>HABITACIONES</v-label>
            </v-col>
            <v-col sm="6" offset-sm="3">
              <v-select :items="filters.bedroomsCount" v-model="selectedFilters.minBedroomsCount"></v-select>
            </v-col>
          </v-row>

          <v-row>
            <v-col cols="12" align="center">
              <v-label>BAÑOS</v-label>
            </v-col>
            <v-col sm="6" offset-sm="3">
              <v-select :items="filters.bathroomsCount" v-model="selectedFilters.minBathroomsCount"></v-select>
            </v-col>
          </v-row>

          <v-row>
            <v-col cols="12" align="center">
              <v-btn color="purple darken-2" dark @click="emitSearchWithFiltersClick">Ver Inmuebles</v-btn>
            </v-col>
          </v-row>
        </v-container>
      </v-form>
    </v-card>
  </v-expand-transition>
</template>

<script>
import lodash from "lodash";

export default {
  name: 'FilterTab',
  methods: {
    emitSearchWithFiltersClick: function() {
      this.$emit('searchWithFiltersClick', this.selectedFilters);
    }
  },
  computed: {
    maxPriceFilter: function() {
      return this.filters.price.filter(price => price.value >= this.selectedFilters.minPrice);
    },
    maxSqmFilter: function() {
      return this.filters.sqm.filter(sqm => sqm.value >= this.selectedFilters.minSqm);
    },
  },
  data: () => ({
    selectedFilters: {
      minPrice: null,
      maxPrice: null,
      minSqm: null,
      maxSqm: null,
      minBedroomsCount: null,
      minBathroomsCount: null
    },
    filters: {
      price: lodash.range(0, 1000001, 25000).map(function(item) {
        return {
          text: `${item} €`,
          value: item
        }
      }),
      sqm: [0,20,25,30,35,...lodash.range(40, 301, 10)].map(function(item) {
        return {
          text: `${item} m²`,
          value: item
        }
      }),
      bedroomsCount: [0,1,2,3,4,5].map(function(count) {
        if(count === 0) {
          return {
            text: "Todos",
            value: null
          }
        }  else {
          return {
            text: `${count}+`,
            value: count
          } 
        }
      }),
      bathroomsCount: [0,1,2,3,4,5].map(function(count) {
        if(count === 0) {
          return {
            text: "Todos",
            value: null
          }
        }  else {
          return {
            text: `${count}+`,
            value: count
          } 
        }
      })
    }
  })
};
</script>