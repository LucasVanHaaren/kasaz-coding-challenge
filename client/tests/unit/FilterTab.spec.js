import { shallowMount, mount } from '@vue/test-utils'
import FilterTab from '@/components/FilterTab.vue'

describe('FilterTab.vue', () => {

  it('match snapshot', () => {
    const wrapper = shallowMount(FilterTab);
    expect(wrapper.element).toMatchSnapshot();
  });

  // it('emit event correctly', () => {
  //   const wrapper = shallowMount(FilterTab);
  //   expect(wrapper.find('button').exists()).toBe(true);
  // });

})
