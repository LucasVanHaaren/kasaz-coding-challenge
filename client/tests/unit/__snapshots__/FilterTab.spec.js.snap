// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterTab.vue match snapshot 1`] = `
<v-expand-transition>
  <v-card
    outlined=""
  >
    <v-form>
      <v-container>
        <v-row>
          <v-col
            align="center"
            cols="12"
          >
            <v-label>
              PRECIO
            </v-label>
          </v-col>
           
          <v-col>
            <v-row>
              <v-select
                items="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
                label="min"
                outlined=""
              />
               
              <v-select
                items="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
                label="max"
                outlined=""
              />
            </v-row>
          </v-col>
        </v-row>
         
        <v-row>
          <v-col
            align="center"
            cols="12"
          >
            <v-label>
              TAMAÑO
            </v-label>
          </v-col>
           
          <v-col>
            <v-row>
              <v-select
                items="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
                label="min"
                outlined=""
              />
               
              <v-select
                items="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
                label="max"
                outlined=""
              />
            </v-row>
          </v-col>
        </v-row>
         
        <v-row>
          <v-col
            align="center"
            cols="12"
          >
            <v-label>
              HABITACIONES
            </v-label>
          </v-col>
           
          <v-col
            offset-sm="3"
            sm="6"
          >
            <v-select
              items="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
            />
          </v-col>
        </v-row>
         
        <v-row>
          <v-col
            align="center"
            cols="12"
          >
            <v-label>
              BAÑOS
            </v-label>
          </v-col>
           
          <v-col
            offset-sm="3"
            sm="6"
          >
            <v-select
              items="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
            />
          </v-col>
        </v-row>
         
        <v-row>
          <v-col
            align="center"
            cols="12"
          >
            <v-btn
              color="purple darken-2"
              dark=""
            >
              Ver Inmuebles
            </v-btn>
          </v-col>
        </v-row>
      </v-container>
    </v-form>
  </v-card>
</v-expand-transition>
`;
